### ISSUE TITLE
mmap on reserved-memory blocks the program

### ISSUE THREAD

**Issue by Ryose** [polarfire-soc/meta-polarfire-soc-yocto-bsp#57]
Hi, I'm using the Yocto-BSP for the enclustra-mpfs SoM wich mounts a PolarFire SoC device. 
Hi have a node in my device-tree which is used to reserve a set of cached memory: 
```
/{
    reserved-memory {
    #address-cells = <2>;
    #size-cells = <2>;
[mem-prova.zip](https://github.com/user-attachments/files/15788146/mem-prova.zip)
    ranges;
    
        reserved: buffer@0 {
            compatible = "shared-dma-pool";
            no-map;
            reg = <0x0 0xC0000000 0x0 0x10000000>;
        };
    };
    
    reserved-driver@0 {
    compatible = "ingeniars,gpu4sat-memory";
    memory-region = <&reserved>;
    };
};
```
My kernel module uses the DMA API to reserve this memory area using a dma_alloc_coherent() and performs a mmap() to give access to a user application, which I attached. Unfortunately, my application blocks indefinitely. 
[mem-prova.zip](https://github.com/user-attachments/files/15788205/mem-prova.zip)

I also attach the mss memory partition configuration of my project and the output of the execution of my application. 
Thanks for your help in advance.
![image](https://github.com/polarfire-soc/meta-polarfire-soc-yocto-bsp/assets/55529780/33820431-8a96-45f6-8579-e7d372a95b16)
![image](https://github.com/polarfire-soc/meta-polarfire-soc-yocto-bsp/assets/55529780/823c6767-1759-420a-bc8b-9bc87534e8bf)

**Comment by eppidei** [comment]
Hi @Ryose , If I understand well the memory gets allocated but the DMA operation stops. Is that correct?  Have you specified the DMA ranges in the device tree?

**Comment by Ryose** [comment]
Yes, I thinks that the allocation is performed correctly( since no error is generated) but when I access said memory region the process stalls.  As you can see, I have declared the node of reserved memory inside my device-tree, is there any additional node that I need to add to my device-tree?

**Comment by vfalanis** [comment]
Closing this issue as it is quite old and may no longer be relevant. This repository will be archived soon in favor of [meta-mchp](https://github.com/linux4microchip/meta-mchp). Please try the latest available release there. If the issue still persists, feel free to create a new ticket. Thank you.
